tests/unit/test_parse.py

```diff
@@ -150,5 +155,59 @@ def test_webhook_missing_keys():
         "workflow_job": {"id": 22428484402, "labels": FAKE_LABELS},
     }
     with pytest.raises(ParseError) as exc_info:
-        webhook_to_job(payload)
+        webhook_to_job(payload, DEFAULT_SELF_HOSTED_LABELS)
     assert f"url key not found in {payload}" in str(exc_info.value)
+
+
+@pytest.mark.parametrize(
+    "labels, ignore_labels",
+    [
+        pytest.param(["self-hosted", "linux", "arm64"], set(), id="empty"),
+        pytest.param(["ubuntu-latest"], {"self-hosted"}, id="non overlapping"),
+        pytest.param(["self-hosted", "linux", "amd"], {"self-hosted", "linux"}, id="overlapping"),
```

Comment:
For this general case, perhaps add more cases where the overlapping is in different positions on the `labels` list. This checks for bugs around ignoring certain elements etc